src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -1,43 +1,394 @@
 
+import com.ctre.phoenix.ErrorCode;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.DemandType;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.kauailabs.navx.frc.AHRS;
 
-import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.networktables.EntryListenerFlags;
+import edu.wpi.first.networktables.EntryNotification;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants;
+import edu.wpi.first.math.MathUtil;
+import frc.robot.Constants.DriveTrainConstants;
+import frc.robot.Constants.TrajectoryConstants;
+
+/**
+ * DriveTrainSubsystem
+ */
+public class DriveTrainSubsystem extends SubsystemBase {
+
+  private final WPI_TalonFX leftMaster = new WPI_TalonFX(DEVICE_ID_LEFT_MASTER);
+  private final WPI_TalonFX leftSlaveOne = new WPI_TalonFX(DEVICE_ID_LEFT_SLAVE_ONE);
+  private final WPI_TalonFX rightMaster = new WPI_TalonFX(DEVICE_ID_RIGHT_MASTER);
+  private final WPI_TalonFX rightSlaveOne = new WPI_TalonFX(DEVICE_ID_RIGHT_SLAVE_ONE);
+
+  private final AHRS gyro = new AHRS(SPI.Port.kMXP);
+  private final DifferentialDriveOdometry differentialDriveOdometry;
+  private Pose2d savedPose;
 
-public class DrivetrainSubsystem extends SubsystemBase {
+  private SlewRateLimiter speedRateLimiter = new SlewRateLimiter(SPEED_RATE_LIMIT_ARCADE);
+  private SlewRateLimiter rotationRateLimiter = new SlewRateLimiter(ROTATE_RATE_LIMIT_ARCADE);
 
-  private final WPI_TalonFX leftLeader = new WPI_TalonFX(LEFT_LEADER_ID);
-  private final WPI_TalonFX leftFollower = new WPI_TalonFX(LEFT_FOLLOWER_ID);
-  private final WPI_TalonFX rightLeader = new WPI_TalonFX(RIGHT_LEADER_ID);
-  private final WPI_TalonFX rightFollower = new WPI_TalonFX(RIGHT_FOLLOWER_ID);
-  private final DifferentialDrive m_Drive = new DifferentialDrive(leftLeader, rightLeader);
+  private boolean useEncoders;
+  private boolean encodersAvailable;
```

Comment:
We can remove all of the `useEncoders` code since the encoders will never fail with Falcon 500/Talon FX. (Well, they can fail, but then the motor will fail as well.)
This code was here because in 2018 and 2019 we had trouble with out drivetrain encoders failing, so we were nervous to rely on them to drive without a fallback.